bitcoin/src/address/mod.rs

```diff
@@ -412,18 +364,17 @@ impl Address {
     ///
     /// This is the preferred non-witness type address.
     #[inline]
-    pub fn p2pkh(pk: impl Into<PubkeyHash>, network: Network) -> Address {
+    pub fn p2pkh(pk: impl Into<PubkeyHash>, network: impl Into<NetworkKind>) -> Address {
```

Comment:
Makes me a bit worried that if someone invents a new network and decides to also add address prefix the whole thing breaks.

Reply:
It's hard to imagine that. p2pkh addresses have been basically deprecated for 8+ years now. And no existing test networks have changed this prefix.

Reply:
That's why it's only a bit. :) I think this one is acceptable.